### PR TITLE
Issue #SB-16395 fix: Filters and group in textbook cards

### PR DESCRIPTION
**Pull request by rahulshukla**

### CODE REVIEW

File: src/app/client/src/app/modules/cbse-program/components/collection/collection.component.ts [hunk 1]
```diff
@@ -67,15 +68,23 @@ export class CollectionComponent implements OnInit, OnDestroy {
...
-    this.subjects = _.find(this.collectionComponentConfig.config.filters.explicit, {'code': 'subject'}).range;
+    this.mediums = _.find(this.collectionComponentConfig.config.filters.implicit, {'code': 'medium'}).defaultValue;
     this.board = _.find(this.collectionComponentConfig.config.filters.implicit, {'code': 'board'}).defaultValue;
+    (_.size(this.mediums) > 1) ? this.isMediumClickable = true : this.isMediumClickable = false;
   }
 
+  getImplecitFilters(): string[] {
+    const sharedcontext = this.collectionComponentInput.programContext.config.sharedContext,
+    implicitFilter = this.collectionComponentConfig.config.filters.implicit,
+    avilableFileters = this.filterByCollection(implicitFilter, 'code', sharedcontext);
```
Comment (venkateshwarans) [inline src/app/client/src/app/modules/cbse-program/components/collection/collection.component.ts]: 'available' and 'getImplecitFilters' spelling

File: src/app/client/src/app/modules/cbse-program/components/collection/collection.component.ts [hunk 2]
```diff
@@ -161,21 +167,23 @@ export class CollectionComponent implements OnInit, OnDestroy {
     return _.groupBy(collection, arg);
   }
 
-  filterCollectionList(filterValue?: string, filterby = 'gradeLevel') {
-    let filterValueItem = _.find(this.collectionComponentConfig.config.filters.explicit, {'code': filterby}).defaultValue;
-    const filterArray = [];
-    if (filterValue) {
+  filterCollectionList(filterValue?: any, filterby = 'gradeLevel') {
+    let filterValueItem: any[];
+    if (Array.isArray(filterValue)) {
```
Comment (venkateshwarans) [inline src/app/client/src/app/modules/cbse-program/components/collection/collection.component.ts]: lets use lodash everywhere

File: src/app/client/src/app/modules/cbse-program/components/collection/collection.component.ts (hunk 1 above)
Comment (rahulshukla) [reply src/app/client/src/app/modules/cbse-program/components/collection/collection.component.ts]: Changed.

File: src/app/client/src/app/modules/cbse-program/components/collection/collection.component.ts (hunk 2 above)
Comment (rahulshukla) [reply src/app/client/src/app/modules/cbse-program/components/collection/collection.component.ts]: Sure, changed

**Pull request merged**